index/scorch/merge.go

```diff
@@ -132,37 +132,54 @@ func (s *Scorch) planMergeAtSnapshot(ourSnapshot *IndexSnapshot) error {
 			if segSnapshot, ok := planSegment.(*SegmentSnapshot); ok {
 				oldMap[segSnapshot.id] = segSnapshot
 				if zapSeg, ok := segSnapshot.segment.(*zap.Segment); ok {
-					segmentsToMerge = append(segmentsToMerge, zapSeg)
+					if segSnapshot.LiveSize() == 0 {
+						oldMap[segSnapshot.id] = nil
+					} else {
+						segmentsToMerge = append(segmentsToMerge, zapSeg)
+					}
 					docsToDrop = append(docsToDrop, segSnapshot.deleted)
```

Comment:
wondering if the maintaining the docsToDrop needs to be moved into line 138.5?

That is, segmentsToMerge and docsToDrop need to have the same lengths... and the same positional pairwise entries?

Reply:
agree with steve, if segmentsToMerge and docsToDrop are not the same lenth, merge will do the wrong thing.